File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left, right = 0, len(heights) - 1
        area = 0
        max_area = 0
        while (left < right):
            width = min(heights[left], heights[right])
            area = min(heights[left], heights[right]) * (right - left)
            if area > max_area:
                max_area = area
            if width == heights[left]:
                left += 1
            else:
                right -= 1
        return max_area

File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-0.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        int n = cost.size();

        if (n == 0) return 0;
        if (n == 1) return cost[0];
        if (n == 2) return min(cost[0], cost[1]);

        vector<int> total_costs(n);

        total_costs[n-1] = cost[n-1];
        total_costs[n-2] = cost[n-2];

        for (int i = n-3; i >= 0; i--) {
            total_costs[i] = cost[i] + min(total_costs[i+1], total_costs[i+2]);
        }

        return min(total_costs[0], total_costs[1]);
    }
};
